### PR TITLE
test: add tests for trident swaps

### PR DESCRIPTION
**Pull request by bayological**
Test cases:
- Single pool: swap ETH from wallet to BAT wallet
- Complex path: swap ETH from wallet to USDT wallet
- Single path: swap BAT from wallet to USDT wallet

### CODE REVIEW

File: src/features/trident/create/classic/SelectAssetsWidget.tsx [hunk 1]
```diff
@@ -17,6 +17,7 @@ const SelectPanel: FC<{ index: number }> = ({ index }) => {
       header={(props) => <SwapAssetPanel.Header {...props} label={i18n._(t`Withdraw from`)} />}
       walletToggle={(props) => (
         <SwapAssetPanel.Switch
+          id={'switch-classic-withdraw-from-' + index}
```
Comment (chillichelli) [inline src/features/trident/create/classic/SelectAssetsWidget.tsx:20]: its better to use ```{`switch-classic-withdraw-from-${index}`}``` as its more readable

File: src/features/trident/pools/DiscoverHeader.tsx [hunk 2]
```diff
@@ -7,9 +7,10 @@ import Button from '../../../components/Button'
 import Typography from '../../../components/Typography'
 import { TridentHeader } from '../../../layouts/Trident'
 
-const HeaderButton: FC<{ title: string; linkTo: string }> = ({ title, linkTo }) => (
+const HeaderButton: FC<{ title: string; linkTo: string; id?: string }> = ({ title, linkTo, id }) => (
   <Link href={linkTo}>
```
Comment (chillichelli) [inline src/features/trident/pools/DiscoverHeader.tsx:11]: need to add passHref={true} here

File: src/pages/trident/swap/index.tsx [hunk 3]
```diff
@@ -163,6 +163,11 @@ const Swap = () => {
                 </div>
               )}
               {isWrap ? <WrapButton /> : <SwapButton />}
+              <input
+                id="trade-type"
+                type="hidden"
+                value={!trade ? '' : trade instanceof Trade ? 'legacy' : 'trident'}
```
Comment (chillichelli) [inline src/pages/trident/swap/index.tsx:169]: use `getTradeVersion(trade) === TradeVersion.V3` function to check if legacy or trident

Comment (bayological) [reply src/pages/trident/swap/index.tsx:169]: Actually don't need this anymore. The latest has the trade type displayed now so will remove and use that.

File: src/features/trident/pools/DiscoverHeader.tsx (hunk 2 above)
Comment (bayological) [reply src/features/trident/pools/DiscoverHeader.tsx:11]: added

**Review by chillichelli — APPROVED** [review]

**Pull request merged**